filebeat/templates/deployment.yaml

```diff
@@ -0,0 +1,141 @@
+# Deploy singleton instance in the whole cluster for some unique data sources, like aws input
+{{- if .Values.deployment.enabled }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: '{{ template "filebeat.fullname" . }}'
+  labels:
+    app: '{{ template "filebeat.fullname" . }}'
+    chart: '{{ .Chart.Name }}-{{ .Chart.Version }}'
+    heritage: '{{ .Release.Service }}'
+    release: '{{ .Release.Name }}'
+  {{- if .Values.deployment.annotations}}
+  annotations:
+    {{- range $key, $value := .Values.deployment.annotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+  {{- end }}
+spec:
+  replicas: {{ .Values.replicas }}
+  selector:
+    matchLabels:
+      app: '{{ template "filebeat.fullname" . }}'
+      heritage: '{{ .Release.Service }}'
```

Comment:
Can you remove `heritage` label from selector? We removed it everywhere because it can be cause of trouble when migrating from Helm 2 to Helm 3